ocs_sample_library_preview/Assets.py

```diff
@@ -0,0 +1,523 @@
+import json
+from typing import List
+
+from .BaseClient import BaseClient
+from .Asset.Data.DataResults import DataResults
+from .Asset.Resolved.ResolvedAsset import ResolvedAsset
+from .Asset.Status.StatusData import StatusData
+from .Asset.Asset import Asset
+from .Asset.AssetType import AssetType
+
+
+class Assets(object):
+    """
+    Client for interacting with OCS Assets
+    """
+
+    def __init__(self, client: BaseClient):
+        """
+        Initializes the Assets client
+        :param client: This is the base client that is used to make REST calls
+        """
+        self.__base_client = client
+        self.__setPathAndQueryTemplates()
+
+    def getAssetById(self, namespace_id: str, asset_id: str) -> Asset:
+        """
+        Returns the specified asset
+        :param namespace_id: The namespace identifier
+        :param asset_id: The asset identifier
+        """
+        if namespace_id is None:
+            raise TypeError
+        if asset_id is None:
+            raise TypeError
+
+        response = self.__base_client.request('get', self.__asset_path.format(
+            namespace_id=namespace_id, asset_id=asset_id))
+        self.__base_client.checkResponse(
+            response, f'Failed to get asset, {asset_id}.')
+
+        result = Asset.fromJson(response.json())
+        return result
+
+    def getAssets(self, namespace_id: str, query: str = '', skip: int = 0, count: int = 100,
+                  ) -> List[Asset]:
+        """
+        Returns a list of assets
+        :param namespace_id: The namespace identifier
+        :param query: An optional query string to search for matching assets.
+        :param skip: An optional parameter representing the zero-based offset of the first asset to
+        retrieve. If not specified, a default value of 0 is used.
+        :param count: An optional parameter, between 1 and 1000 (inclusive), that represents the
+        maximum number of retrieved assets. If not specified, the default is 100.
+        """
+        if namespace_id is None:
+            raise TypeError
+
+        response = self.__base_client.request('get', self.__assets_path.format(
+            namespace_id=namespace_id), params={'skip': skip, 'count': count, 'query': query})
+        self.__base_client.checkResponse(response, f'Failed to get assets.')
+
+        results = []
+        for i in response.json():
+            results.append(Asset.fromJson(i))
+        return results
+
+    def createAsset(self, namespace_id: str, asset: Asset) -> Asset:
+        """
+        Create a new asset with a specified Id
```

Comment:
A specified Id or Asset object?

Reply:
Well, not exactly. The text is copied from the description [here](https://ocs-docs.osisoft.com/Content_Portal/Documentation/Assets/assets-api.html#create-asset). The point of this statement is that usually the asset object doesn't have to specify an Id (if you are pushing an update instead of a create, for example). When you use this endpoint, the asset object has to have an Id specified on the object.